Doc/c-api/float.rst

```diff
@@ -76,3 +76,83 @@ Floating Point Objects
 .. c:function:: double PyFloat_GetMin()
 
    Return the minimum normalized positive float *DBL_MIN* as C :c:type:`double`.
+
+
+Pack and Unpack functions
+=========================
+
+The :mod:`struct` and :mod:`pickle` (at least) modules need an efficient
+platform- independent way to store floating-point values as byte strings. The
+Pack routines produce a bytes string from a C :c:type:`double`, and the Unpack
+routines produce a C :c:type:`double` from such a bytes string. The suffix
+(2, 4 or 8) specifies the number of bytes in the bytes string.
+
+On platforms that appear to use IEEE-754 formats these functions work by
+copying bits. On other platforms, the formats the 4- byte format (32-bit) is
+identical to the IEEE-754 single precision format, and the 8-byte format
+(64-bit) to the IEEE-754 double precision format, although the packing of INFs
+and NaNs (if such things exist on the platform) isn't handled correctly, and
+attempting to unpack a bytes string containing an IEEE INF or NaN will raise an
+exception.
+
+On non-IEEE platforms with more precision, or larger dynamic range, than IEEE
+754 supports, not all values can be packed; on non-IEEE platforms with less
+precision, or smaller dynamic range, not all values can be unpacked. What
+happens in such cases is partly accidental (alas).
+
+.. versionadded:: 3.11
+
+Pack functions
+--------------
+
+The pack routines write 2, 4 or 8 bytes, starting at *p*. *le* is a bool
+argument, true if you want the bytes string in little-endian format (exponent
+last, at ``p+1``, ``p+3`` or ``p+7``), false if you want big-endian format
+(exponent first, at *p*).
+
+Return value: ``0`` if all is OK, ``-1`` if error (and an exception is set,
+most likely :exc:`OverflowError`).
+
+There are two problems on non-IEEE platforms:
+
+* What this does is undefined if *x* is a NaN or infinity.
+* -0.0 and +0.0 produce the same bytes string.
+
+.. c:function:: int PyFloat_Pack2(double x, unsigned char *p, int le)
+
+   Pack a C double as 16-bit.
```

Comment:
Suggest "2 bytes" instead of 16 bits? It's probably worth re-mentioning the IEEE 754 binary16 format here, too. Similar comments apply to the next two entries.